scripts/shared/lib/utils

```diff
@@ -53,52 +53,48 @@ function with_context() {
 }
 
 # Run cluster commands in parallel.
-# 1st argument is the numbers of the clusters to run for, supports "1 2 3" or "{1..3}" for range
+# 1st argument is the cluster names for which to run.
 # 2nd argument is the command to execute, which will have the $cluster variable set.
 # 3rd argument and so forth get passed to the command.
 function run_parallel() {
-    local clusters cmnd
-    clusters=$(eval echo "$1")
-    cmnd=$2
+    local cmnd=$2
     declare -A pids
-    for i in ${clusters}; do
+    for cluster in $(eval echo "$1"); do
         (
            set -o pipefail
-           with_context "cluster${i}" "$cmnd" "${@:3}" | sed "s/^/[cluster${i}] /"
+           $cmnd "${@:3}" | sed "s/^/[${cluster}] /"
         ) &
-        pids["${i}"]=$!
+        pids["${cluster}"]=$!
     done
 
-    for i in "${!pids[@]}"; do
-        wait ${pids[$i]}
+    for pid in "${pids[@]}"; do
+        wait $pid
     done
 }
 
 # Run the given command (with any arguments) in parallel on all the clusters.
 function run_all_clusters() {
-    run_parallel "{1..3}" "$@"
+    run_parallel "${clusters[*]}" "$@"
```

Comment:
```suggestion
    run_parallel "${clusters[@]}" "$@"
```

Reply:
Using the '@' syntax doesn't work.

Reply:
Ah of course, `run_parallel` needs its first argument to contain all the targets...